test/TestConfig/scripts/testKitGen/makeGenTool/mkgen.pl

```diff
@@ -673,6 +662,29 @@ sub writeTargets {
 							my $lgtKey = $eachLevel . '.' . $eachGroup . '.' . $eachType;
 							push(@{$groupTargets{$lgtKey}}, $name);
 						}
+						else {
+							my $lgtKey = $eachLevel . '.' . $eachGroup . '.' . $eachType;
+							my $dlgtKey = 'disabled.' . $lgtKey;
+							my $echodlgtKey = 'echo.' . $dlgtKey;
+							push(@{$groupTargets{$dlgtKey}}, $name);
+							push(@{$groupTargets{$echodlgtKey}}, "echo.disabled." . $name);
+							print $fhOut "echo.disabled." . $name . ":\n";
+							print $fhOut "$indent\@echo \"\" | tee -a \$(Q)\$(TESTOUTPUT)\$(D)TestTargetResult\$(Q);\n";
+							print $fhOut "$indent\@echo \"===============================================\" | tee -a \$(Q)\$(TESTOUTPUT)\$(D)TestTargetResult\$(Q);\n";
+							print $fhOut "$indent\@echo \"Running test $name ...\" | tee -a \$(Q)\$(TESTOUTPUT)\$(D)TestTargetResult\$(Q);\n";
+							print $fhOut "$indent\@echo \"===============================================\" | tee -a \$(Q)\$(TESTOUTPUT)\$(D)TestTargetResult\$(Q);\n";
+							print $fhOut "$indent\@perl \'-MTime::HiRes=gettimeofday\' -e \'print \"$name Start Time: \" . localtime() . \" Epoch Time (ms): \" . int (gettimeofday * 1000) . \"\\n\"\' | tee -a \$(Q)\$(TESTOUTPUT)\$(D)TestTargetResult\$(Q);\n";
+							print $fhOut "$indent\@echo \"" . $name . "_DISABLED\" | tee -a \$(Q)\$(TESTOUTPUT)\$(D)TestTargetResult\$(Q);\n";
+							print $fhOut "$indent\@echo \"Disabled Reason:\"\n";
+							my @disabledReasons = split("[\t\n]", $test->{'disabled'});
+							foreach my $dReason (@disabledReasons) {
+								if ($dReason ne "") {
+									print $fhOut "$indent\@echo \"$dReason\"\n";
```

Comment:
Add `| tee -a \$(Q)\$(TESTOUTPUT)\$(D)TestTargetResult\$(Q);` in the the print out.